src/executor_scheme_redirect.c

```diff
@@ -0,0 +1,118 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include "debugger.h"
+#include "libft.h"
+
+#include "filesystem.h"
+#include "ft_unistd.h"
+#include "process.h"
+#include "execscheme.h"
+#include "executor.h"
+
+#define CHILD 0
+
+static t_bool	redirection_write_handle_fd(const char *fname, t_bool should_append)
+{
+	int			fd;
+	t_bool		success;
+
+	assert(fname);
+	success = FALSE;
+	if (should_append)
+		fd = open(fname, O_WRONLY | O_CREAT | O_APPEND, 0644);
+	else
+		fd = open(fname, O_WRONLY | O_CREAT | O_TRUNC, 0644);
+		
+	if (fd != -1)
+	{
+		success = (dup2(fd, STDOUT) != -1);
+		close(fd);
+	}
+	return (success);
+}
+
+static t_bool	redirection_read_handle_fd(const char *fname)
+{
+	int			fd;
+	t_bool		success;
+
+	assert(fname);
+	success = FALSE;
+	fd = fs_open(fname, O_RDONLY);
+	if (fd != -1)
+	{
+		success = (dup2(fd, STDIN) != -1);
+		close(fd);
+	}
+	return (success);
+}
+
+int			handler_scheme_redirection(t_execscheme *scheme)
+{
```

Comment:
this would need to be splited

Reply:
That's from pullrequest #164  (which I merged from for this one, since they are slightly dependent) but you are totally right! 

the whole execution needs to be refactored

but i figured we better make sure áll the functionality is in before we spend to much time refactoring

I'll merge this one, and when everything runs up to spec, i'll refactor is (contrary to LeBlanc's law haha)